src/main/java/org/prebid/server/bidder/adquery/AdQueryBidder.java

```diff
@@ -0,0 +1,213 @@
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+                continue;
+            }
+            httpRequests.add(createRequest(request, imp, extImpAdQuery));
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private ExtImpAdQuery parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), AD_QUERY_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private HttpRequest<AdQueryRequest> createRequest(BidRequest bidRequest, Imp imp, ExtImpAdQuery extImpAdQuery) {
+        final AdQueryRequest outgoingRequest = createAdQueryRequest(bidRequest, imp, extImpAdQuery);
+
+        return HttpRequest.<AdQueryRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(resolveHeader(bidRequest.getDevice()))
+                .impIds(BidderUtil.impIds(bidRequest))
+                .payload(outgoingRequest)
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .build();
+    }
+
+    private AdQueryRequest createAdQueryRequest(BidRequest bidRequest, Imp imp, ExtImpAdQuery extImpAdQuery) {
+        return AdQueryRequest.builder()
+                .v(PREBID_VERSION)
+                .placementCode(extImpAdQuery.getPlacementId())
+                .auctionId(StringUtils.EMPTY)
+                .type(extImpAdQuery.getType())
+                .adUnitCode(imp.getTagid())
+                .bidQid(ObjectUtil.getIfNotNullOrDefault(bidRequest.getUser(), User::getId, () -> StringUtils.EMPTY))
+                .bidId(String.format("%s%s", bidRequest.getId(), imp.getId()))
+                .bidder(BIDDER_NAME)
+                .bidderRequestId(bidRequest.getId())
+                .bidRequestsCount(1)
+                .bidderRequestsCount(1)
+                .sizes(getImpSizes(imp))
+                .bidIp(ObjectUtil.getIfNotNull(bidRequest.getDevice(), Device::getIp))
+                .bidIpv6(ObjectUtil.getIfNotNull(bidRequest.getDevice(), Device::getIpv6))
+                .bidUa(ObjectUtil.getIfNotNull(bidRequest.getDevice(), Device::getUa))
+                .bidPageUrl(ObjectUtil.getIfNotNull(bidRequest.getSite(), Site::getPage))
+                .build();
+
+    }
+
+    private String getImpSizes(Imp imp) {
+        final Banner banner = imp.getBanner();
+        if (banner == null) {
+            return StringUtils.EMPTY;
+        }
+
+        final List<Format> format = banner.getFormat();
+        if (CollectionUtils.isNotEmpty(format)) {
+            final List<String> sizes = new ArrayList<>();
+            format.forEach(singleFormat -> sizes.add(
+                    "%sx%s".formatted(getIntOrElseZero(singleFormat.getW()), getIntOrElseZero(singleFormat.getH()))));
+            return String.join("_", sizes);
+        }
```

Comment:
it's possible to join in a single stream: `formats.stream().map(...).collect(Collectors.joining("_"))`